### PR TITLE
Add FieldManager to client-go fake

### PR DESCRIPTION
**Pull request by apelisse**
Allow fieldmanager to be wired inside the client-go fake so that we can test server-side apply with it.

#### What type of PR is this?
/kind feature

#### Which issue(s) this PR fixes:
TODO: This is a draft, but this will probably close a bunch of issues.
Fixes #

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/client-go/testing/fixture.go [hunk 1]
```diff
@@ -142,6 +150,28 @@ func ObjectReaction(tracker ObjectTracker) ReactionFunc {
 			}
 			return true, nil, nil
 
+		// Check for Apply before Patch because Apply is made of Patch.
+		case ApplyActionImpl:
+			if action.GetSubresource() != "" {
+				return true, nil, errors.NewBadRequest(fmt.Sprintf("sub-resources are not supported yet."))
```
Comment (mimowo) [inline staging/src/k8s.io/client-go/testing/fixture.go:156]: IIUC, this needs to be required to adjust most of the tests, based on the list of failures: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/116855/pull-kubernetes-unit/1679166711774842880, for example this one: https://github.com/kubernetes/kubernetes/blob/3cbfa3e2cd7e756b1f51a5c7e014c415ecdb1253/pkg/controller/disruption/disruption_test.go#L1424. 

Is it planned to be covered in this PR, or a follow up?

Comment (apelisse) [reply staging/src/k8s.io/client-go/testing/fixture.go:156]: I think I have to fix the tests that broke within this PR.

Comment (apelisse) [reply staging/src/k8s.io/client-go/testing/fixture.go:156]: Faking some of the subresources isn't going to be easy though.

**Comment by dims** [comment]
This work-in-progress PR needs a rebase. please rebase if this PR is still needed for the upcoming release.

File: staging/src/k8s.io/client-go/testing/fixture.go [hunk 2]
```diff
@@ -352,14 +390,96 @@ func (t *tracker) Add(obj runtime.Object) error {
...
+	// We need to get back the same underlying type of object as was
+	// given in, so we need to put the result back into obj. Reset
+	// it first.
+	if err := setObject(obj, newObj.(*unstructured.Unstructured)); err != nil {
+		return err
+	}
+
+	fmt.Printf("Final object after apply: %v\n", obj)
+
+	return t.add(gvr, obj, ns, replace)
```
Comment (howardjohn) [inline staging/src/k8s.io/client-go/testing/fixture.go:468]: Do we need to do diff of old vs new to avoid NOP Apply() calls triggering updates?

**Pull request closed without merging**